pkg/controller/volume/persistentvolume/controller_base.go

```diff
@@ -509,6 +509,29 @@ func (ctrl *PersistentVolumeController) upgradeVolumeFrom1_2(volume *api.Persist
 	return true
 }
 
+// findDeleterParameters finds StorageClass.Parameters of storage class where
+// given volume belongs
+func (ctrl *PersistentVolumeController) findDeleterParameters(volume *api.PersistentVolume) (map[string]string, error) {
+	className := getVolumeClass(volume)
+	if className == "" {
+		return nil, nil
+	}
+
+	obj, found, err := ctrl.classes.GetByKey(className)
```

Comment:
...which is going to mean a different set of changes to the method signatures involved here :-D